### PR TITLE
LUCENE-8638: remove deprecated Legacy*DocValues* classes in org.apache.lucene.codecs.memory package

### PR DESCRIPTION
**Pull request by cpoerschke**
https://issues.apache.org/jira/browse/LUCENE-8638

### CODE REVIEW

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedSetDocValuesWrapper.java [hunk 1]
```diff
@@ -1,113 +0,0 @@
...
-import java.io.IOException;
-import org.apache.lucene.index.SortedSetDocValues;
-import org.apache.lucene.util.BytesRef;
-
-/**
- * Wraps a {@link LegacySortedSetDocValues} into a {@link SortedSetDocValues}.
- *
- * @deprecated Implement {@link SortedSetDocValues} directly.
- */
-@Deprecated
```
Comment (cpoerschke) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedSetDocValuesWrapper.java:29]: annotated as Deprecated

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedSetDocValues.java [hunk 2]
```diff
@@ -1,109 +0,0 @@
...
-/**
- * A per-document set of presorted byte[] values.
- *
- * <p>Per-Document values in a SortedDocValues are deduplicated, dereferenced, and sorted into a
- * dictionary of unique values. A pointer to the dictionary value (ordinal) can be retrieved for
- * each document. Ordinals are dense and in increasing sorted order.
- *
- * @deprecated Use {@link SortedSetDocValues} instead.
- */
-@Deprecated
```
Comment (cpoerschke) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedSetDocValues.java:33]: annotated as Deprecated

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedNumericDocValuesWrapper.java [hunk 3]
```diff
@@ -1,100 +0,0 @@
...
-
-import java.io.IOException;
-import org.apache.lucene.index.SortedNumericDocValues;
-
-/**
- * Wraps a {@link LegacySortedNumericDocValues} into a {@link SortedNumericDocValues}.
- *
- * @deprecated Implement {@link SortedNumericDocValues} directly.
- */
-@Deprecated
```
Comment (cpoerschke) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedNumericDocValuesWrapper.java:28]: annotated as Deprecated

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedNumericDocValues.java [hunk 4]
```diff
@@ -1,46 +0,0 @@
...
-package org.apache.lucene.codecs.memory;
-
-import org.apache.lucene.index.SortedNumericDocValues;
-
-/**
- * A list of per-document numeric values, sorted according to {@link Long#compare(long, long)}.
- *
- * @deprecated Use {@link SortedNumericDocValues} instead.
- */
-@Deprecated
```
Comment (cpoerschke) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedNumericDocValues.java:26]: annotated as Deprecated

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedDocValuesWrapper.java [hunk 5]
```diff
@@ -1,102 +0,0 @@
...
-import java.io.IOException;
-import org.apache.lucene.index.SortedDocValues;
-import org.apache.lucene.util.BytesRef;
-
-/**
- * Wraps a {@link LegacySortedDocValues} into a {@link SortedDocValues}.
- *
- * @deprecated Implement {@link SortedDocValues} directly.
- */
-@Deprecated
```
Comment (cpoerschke) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedDocValuesWrapper.java:29]: annotated as Deprecated

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedDocValues.java [hunk 6]
```diff
@@ -1,111 +0,0 @@
...
-/**
- * A per-document byte[] with presorted values.
- *
- * <p>Per-Document values in a SortedDocValues are deduplicated, dereferenced, and sorted into a
- * dictionary of unique values. A pointer to the dictionary value (ordinal) can be retrieved for
- * each document. Ordinals are dense and in increasing sorted order.
- *
- * @deprecated Use {@link SortedDocValues} instead.
- */
-@Deprecated
```
Comment (cpoerschke) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacySortedDocValues.java:32]: annotated as Deprecated

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacyNumericDocValuesWrapper.java [hunk 7]
```diff
@@ -1,96 +0,0 @@
...
-import java.io.IOException;
-import org.apache.lucene.index.NumericDocValues;
-import org.apache.lucene.util.Bits;
-
-/**
- * Wraps a {@link LegacyNumericDocValues} into a {@link NumericDocValues}.
- *
- * @deprecated Implement {@link NumericDocValues} directly.
- */
-@Deprecated
```
Comment (cpoerschke) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacyNumericDocValuesWrapper.java:29]: annotated as Deprecated

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacyNumericDocValues.java [hunk 8]
```diff
@@ -1,39 +0,0 @@
...
-package org.apache.lucene.codecs.memory;
-
-import org.apache.lucene.index.NumericDocValues;
-
-/**
- * A per-document numeric value.
- *
- * @deprecated Use {@link NumericDocValues} instead.
- */
-@Deprecated
```
Comment (cpoerschke) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacyNumericDocValues.java:26]: annotated as Deprecated

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacyDocValuesIterables.java [hunk 9]
```diff
@@ -1,567 +0,0 @@
...
-import org.apache.lucene.index.BinaryDocValues;
-import org.apache.lucene.index.FieldInfo;
-import org.apache.lucene.index.NumericDocValues;
-import org.apache.lucene.index.SortedDocValues;
-import org.apache.lucene.index.SortedNumericDocValues;
-import org.apache.lucene.index.SortedSetDocValues;
-import org.apache.lucene.util.BytesRef;
-
-/** Bridge helper methods for legacy codecs to map sorted doc values to iterables. */
-class LegacyDocValuesIterables {
```
Comment (cpoerschke) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacyDocValuesIterables.java:36]: _not_ annotated as Deprecated but all public stuff within it _is_ annotated as Deprecated

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacyBinaryDocValuesWrapper.java [hunk 10]
```diff
@@ -1,91 +0,0 @@
...
-import org.apache.lucene.index.BinaryDocValues;
-import org.apache.lucene.util.Bits;
-import org.apache.lucene.util.BytesRef;
-
-/**
- * Wraps a {@link LegacyBinaryDocValues} into a {@link BinaryDocValues}.
- *
- * @deprecated Implement {@link BinaryDocValues} directly.
- */
-@Deprecated
```
Comment (cpoerschke) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacyBinaryDocValuesWrapper.java:30]: annotated as Deprecated

File: lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacyBinaryDocValues.java [hunk 11]
```diff
@@ -1,39 +0,0 @@
...
-
-import org.apache.lucene.index.BinaryDocValues;
-import org.apache.lucene.util.BytesRef;
-
-/**
- * A per-document byte[]
- *
- * @deprecated Use {@link BinaryDocValues} instead.
- */
-@Deprecated
```
Comment (cpoerschke) [inline lucene/codecs/src/java/org/apache/lucene/codecs/memory/LegacyBinaryDocValues.java:27]: annotated as Deprecated

**Review by jpountz — APPROVED** [review]

**Pull request merged**